sql/information_schema/information_schema.go

```diff
@@ -887,6 +892,10 @@ func columnStatisticsRowIter(ctx *Context, c Catalog) (RowIter, error) {
 			}
 
 			for _, col := range t.Schema() {
+				privSetCol := privSetTbl.Column(col.Name)
+				if privSetCount == 0 && privSetDb.Count() == 0 && privSetTbl.Count() == 0 && privSetCol.Count() == 0 {
+					continue
+				}
```

Comment:
Would be nice to have a simple `UserPrivilegeTest` to assert that when there are no privileges, a user can't see info they shouldn't see in `information_schema.column_statistics`.